en/08_Changelogs/6.0.0.md

```diff
@@ -181,10 +183,25 @@ SilverStripe\UserForms\Model\UserDefinedForm:
 
 </details>
 
-### Other new features
+### Changes to the templating/view layer {#view-layer}
 
-- Native indexed PHP arrays can now be passed into templates and iterated over with `<% loop $MyArray %>`. Under the hood they are wrapped in [`ArrayList`](api:SilverStripe\View\ViewableData), so you can get the count using `$Count` and use `<% if $ArrayList %>` as a shortcut for `<% if $ArrayList.Count %>`. Other functionality from `ArrayList` such as filtering and sorting cannot be used on arrays since they don't have keys to filter or sort against.
+#### Strong typing for `ViewableData` and `DBField`
+
+Many of the properties and methods in [`ViewableData`](api:SilverStripe\View\ViewableData), [`DBField`](api:SilverStripe\ORM\FieldType\DBField), and their immediate subclasses have been given strong typehints. Previously, these only had typehints in the PHPDoc which meant that any arbitrary values could be assigned or returned.
+
+Most of the strong types are either identical to the old PHPDoc types, or match what was already actually being assigned to, passed into, and returned from those APIs.
+
+In some cases, where a string was expected but sometimes `null` was being used, we have explicitly strongly typed the API to `string`. This matches [similar changes PHP made in PHP 8.1](https://php.watch/versions/8.1/internal-func-non-nullable-null-deprecation) and will help avoid passing `null` values in to functions that expect a string.
+
+The one change we specifically want to call out is for [`ViewableData::obj()`](api:SilverStripe\View\ViewableData::obj()). This method will now explicitly return `null` if there is no field, property, method, etc to represent the field you're trying to get an object for. This is a change from the old behaviour, where an empty `DBField` instance would be returned even though there's no way for any non-null value to be available for that field.
+
+See the [full list of removed and changed API](#api-removed-and-changed) to see all of the API with updated typing.
+
+### Other new features and small changes {#other-new-features}
+
```

Comment:
Renamed because the note about the default value isn't really a new feature, but there's also no need for it to have its own section.
Since this is a major release changelog it makes sense for things beyond just features to be in this section.